### PR TITLE
Adjusts home directory path on Windows (fixes #1959)

### LINKED ISSUE #1959: Hyper crashes on launch on recent Windows 10 builds
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: Windows 10 Pro Insider Preview, build 16215
- **Hyper.app version**: 1.3.3
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: https://gist.github.com/cerebrate/ac9aa51a9dfa641a23fe1881f3be449a
- **Relevant information from devtools** _(CMD+ALT+I on Mac OS, CTRL+SHIFT+I elsewhere)_: N/A
- **The issue is reproducible in vanilla Hyper.app**: Yes. Removing plugins has no effect on problem.

## Issue
Hyper crashes on launch with the following error:

[Window Title]
Error

[Main Instruction]
A JavaScript error occurred in the main process

[Content]
Uncaught Exception:
Error: EINVAL: invalid argument, lstat 'C:\Users\avatar\OneDrive'
    at Error (native)
    at fs.lstatSync (fs.js:982:18)
    at Object.fs.lstatSync (ELECTRON_ASAR.js:244:16)
    at Gaze._addToWatched (C:\Users\avatar\AppData\Local\hyper\app-1.3.3\resources\app\node_modules\gaze\lib\gaze.js:292:14)
    at Gaze.add (C:\Users\avatar\AppData\Local\hyper\app-1.3.3\resources\app\node_modules\gaze\lib\gaze.js:190:8)
    at new Gaze (C:\Users\avatar\AppData\Local\hyper\app-1.3.3\resources\app\node_modules\gaze\lib\gaze.js:74:10)
    at gaze (C:\Users\avatar\AppData\Local\hyper\app-1.3.3\resources\app\node_modules\gaze\lib\gaze.js:86:10)
    at watch (C:\Users\avatar\AppData\Local\hyper\app-1.3.3\resources\app\config.js:44:3)
    at Object.exports.init (C:\Users\avatar\AppData\Local\hyper\app-1.3.3\resources\app\config.js:130:3)
    at Object.<anonymous> (C:\Users\avatar\AppData\Local\hyper\app-1.3.3\resources\app\index.js:62:8)

[OK]

I believe this may relate to the changes to OneDrive in Windows 10 build 16215 and higher, in which placeholders are once again available in place of conventional files, although I don't know why Hyper is attempting to access that folder in the first place.

### PR DESCRIPTION
**Pull request by cerebrate**
Adjusts the home directory path (source of the config file path) on Win32 to the AppData\Roaming\Hyper subfolder of the folder returned by homedir(). This fixes issue #1959 by eliminating the OneDrive subdirectory from those watched by gaze.

(This also puts storage of the Hyper configuration in the Windows design guidelines recommended location, enables it to roam with the user when roaming profiles are in use, and should improve system performance impact by not watching all the potentially many thousands of files under homedir() not relevant to Hyper configuration, but the fix to #1959 is the important thing.)

Should be ready to merge.

### CODE REVIEW

**Comment by cerebrate** [comment]
This should also fix #1944 etal.

File: app/config/paths.js [hunk 1]
```diff
@@ -6,7 +6,11 @@ const isDev = require('electron-is-dev');
 
 const cfgFile = '.hyper.js';
 const defaultCfgFile = 'config-default.js';
-const homeDir = homedir();
+let homeDir = homedir();
+
+if (process.platform === 'win32') {
+  homeDir = resolve(homeDir, 'AppData\\Roaming\\Hyper');
```
Comment (nesl247) [inline app/config/paths.js]: Any reason to not use `app.getPath('userData')`? That actually solves this issue, but also adds support for XDG_CONFIG_HOME and such.

Comment (nesl247) [inline app/config/paths.js]: With this in mind, https://github.com/zeit/hyper/pull/170 I believe would also fix the issue in this PR, as well as add the support for XDG_CONFIG_HOME, the original config path, etc.

**Comment by nesl247** [comment]
https://github.com/zeit/hyper/pull/1946 would also fix the issue but solving the root cause.

**Comment by cerebrate** [comment]
@nesl247 No reason not to use ```app.getPath('userData')``` that I'm aware of, and indeed, it would be a better fix to do so, knowing that it's there. (Node.js in general isn't my first language, alas.)

( FWIW, I saw #1595 and #170 , but both appear to have been sitting around for a while. I was looking to provide a minimal no-feature/no-muss fix with this PR in the hope of quick review/merge, since Hyper is currently broken without one on Windows builds 16215+.)

**Comment by DuIslingr** [comment]
Note that hyper only breaks on those builds if the onedrive on demand feature is enabled. otherwise it works fine still as is.

**Comment by colbin8r** [comment]
Can we get this merged? Hyper is broken for me due to this.

**Comment by UsmanMohammad** [comment]
If anyone is waiting for the branch to be accepted, you can clone @cerebrate 's branch and build. A few prerequisites : 

- You will need the latest version of node, use nvm for windows which allows you to switch the version of node you're running (you will need to uninstall node and delete the folders for nodejs and npm in order for nvm to work properly) 
- Ensure python 2 is installed and a version of visual studio is installed (2017 community is free)
- Let npm know the version of visual studio it will use to build node-pty `npm config set msvs_version <year>`
- Follow build instructions for hyper

**Comment by leo** [comment]
Should be fixed with https://github.com/zeit/hyper/pull/1946? Can someone check?

**Comment by DuIslingr** [comment]
@leo No that did not solve the issue. It still occurred unless I turned off the mentioned setting in OneDrive, then hyper proceeded to work normally.

**Comment by chabou** [comment]
I'm agree: `app.getPath('userData')` is better.
And this PR can fix #1959 until #1595 has been rebased/merged (it can take a long time).

But we should manage path transition. We probably should move current config file in the new location and maybe notify user about this.

**Comment by chabou** [comment]
There are some discussions about this in #1595.
Any case, we can't use another place for config file without current config migration.

BTW thank you so much for your help and really sorry for answer delays 🙏

**Pull request closed without merging**